src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Logging/ActivityLogLoggerProvider.Logger.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.CodeAnalysis.Razor.Logging;
+using Microsoft.VisualStudio.Razor.Logging;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Logging;
+
+internal sealed partial class ActivityLogLoggerProvider
+{
+    private sealed class Logger(RazorActivityLog activityLog, string categoryName) : ILogger
+    {
+        private readonly RazorActivityLog _activityLog = activityLog;
+        private readonly string _categoryName = categoryName;
+
+        public bool IsEnabled(LogLevel logLevel)
+            => logLevel is LogLevel.Warning or LogLevel.Error or LogLevel.Critical;
+
+        public void Log(LogLevel logLevel, string message, Exception? exception)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.Error or LogLevel.Critical:
```

Comment:
nit: We usually would do something like 

```c# 
case LogLevel.Error:
case LogLevel.Critical:
``` 

It doesn't matter here, nor am I sure what I like better, but just acknowledging my brain noticed this :P

Reply:
I definitely understand switch-case fall through, but I'm not sure I agree that it's an established style. I expect it's just more habitual, since pattern-matching is clearly allowed in switch statements and we use it there.